scripts/process/clean_county_boundaries.R

```diff
@@ -0,0 +1,255 @@
+    base_no_ext <- gsub('\\.zip', '', basename(zipname))
+    year <- substring(base_no_ext, nchar(base_no_ext)-3)
+    list(files=files, year=year)
+  })
+  names(map_shps) <- sapply(map_shps, function(map_shp) map_shp$year)
+  
+  # load into R
+  decade_shps <- lapply(map_shps, function(shapefile_files) {
+    message("reading county data for year ", shapefile_files$year)
+    shp_file <- grep("\\.shp$", shapefile_files$files, value=TRUE)
+    layer <- sf::st_layers(shp_file)$name
+    one_year_sf <- sf::st_read(shp_file, layer=layer, stringsAsFactors=FALSE, quiet=TRUE, check_ring_dir=TRUE)
+    list(sf=one_year_sf, year=shapefile_files$year)
+  })
+  
+  # harmonize each decadal dataset and combine them into 1
+  all_shps_simple <- simplify_combine_shps(decade_shps)
+  
+  # supplement state info with names and abbreviations from dataRetrieval and
+  # our made-up abbreviations for the territories of AK and HI
+  states <- consolidate_state_info(all_shps_simple)
+  
+  # supplement county info with long names from dataRetrieval. 214 counties are
+  # missing either a long or short name even after this merge, and 16 of those
+  # have neither. fill these in as best we can - reuse long or short name if
+  # available, otherwise use the county FIPS
+  counties <- consolidate_county_info(all_shps_simple)
+  
+  # split the country-wide shapefiles into state-wide shapefiles
+  split_shps <- lapply(setNames(nm=states$state_FIPS[c(1,4)]), function(state_fips) {
+    message('splitting out shapefiles for state ', state_fips)
+    
+    # subset to just one state
+    state_shp <- all_shps_simple %>%
+      filter(state_FIPS == state_fips) %>%
+      select(year, state_FIPS, county_FIPS, geometry)
+    
+    # # consolidate into a minimal number of non-duplicated county boundaries
+    # message('consolidating ', states %>% filter(state_FIPS==state_fips) %>% pull(state_name))
+    # distinct_shps <- consolidate_county_polygons(state_shp)
+    # return(distinct_shps)
+    
+    return(state_shp)
+  })
+  
+  geojsondir <- file.path(tempdir(), 'geojson')
+  if(!dir.exists(geojsondir)) dir.create(geojsondir)
+  lapply(setNames(nm=names(split_shps)), function(split_shp_nm) {
+    outfile <- file.path(geojsondir, sprintf('%s.geojson', split_shp_nm))
+    message('writing ', outfile)
+    split_shp <- split_shps[[split_shp_nm]]
+    geojsonio::geojson_write(
+      input=split_shp,
+      file=outfile,
+      geometry='polygon',
+      convert_wgs84=TRUE)
+  })
+  # also write the states and counties tables
+  jsonlite::write_json(states, file.path(geojsondir, 'states.json'))
+  jsonlite::write_json(counties, file.path(geojsondir, 'counties.json'))
+  
+  # save to one big zip file
+  oldwd <- setwd(geojsondir)
+  on.exit(setwd(oldwd))
+  zipfile <- file.path(oldwd, viz[['location']])
+  if(file.exists(zipfile)) file.remove(zipfile)
+  filestozip <- dir()
+  zip(zipfile, files=filestozip)
+}
+
+simplify_combine_shps <- function(decade_shps) {
```

Comment:
I'm confused about the use of the word simplify here. This is just simplifying attributes right?

Reply:
yes, simplifying attributes here.

Reply:
Yeah. The buffer usually fixes self intersection errors, but it only goes so far. It wasn't terrible to install on mac, but I believe it would be hard on windows...